rclpy/src/rclpy/_rclpy.c

```diff
@@ -2587,12 +2588,12 @@ rclpy_wait_set_add_entity(PyObject * Py_UNUSED(self), PyObject * args)
     ret = rcl_wait_set_add_event(wait_set, event, &index);
   } else {
     ret = RCL_RET_ERROR;  // to avoid a linter warning
-    PyErr_Format(PyExc_RuntimeError,
+    PyErr_Format(RCLError,
```

Comment:
Not an error from `rcl`.
```suggestion
    PyErr_Format(PyExc_RuntimeError,
```